Source/DafnyCore/Verifier/Translator.ExpressionWellformed.cs

```diff
@@ -895,6 +897,37 @@ private class WFOptions {
                 builder.Add(Assert(GetToken(expr), Bpl.Expr.Le(Bpl.Expr.Literal(0), etran.TrExpr(e.E1)), positiveDesc, options.AssertKv));
                 builder.Add(Assert(GetToken(expr), Bpl.Expr.Le(etran.TrExpr(e.E1), Bpl.Expr.Literal(w)), upperDesc, options.AssertKv));
               }
+            }
+            break;
+          case BinaryExpr.ResolvedOpcode.EqCommon:
+          case BinaryExpr.ResolvedOpcode.NeqCommon:
+            CheckWellformed(e.E1, options, locals, builder, etran);
+            if (e.InCompiledContext) {
+              if (Resolver.CanCompareWith(e.E0) || Resolver.CanCompareWith(e.E1)) {
+                // everything's fine
+              } else if (!e.E0.Type.SupportsEquality) {
```

Comment:
You mean on L908, right? (For L906, see my comment above.) The test on L908 is unnecessary, because `CanCompareWith` on L906 always returns `true` if `.SupportsEquality` holds. (This applies to both `e.E0` and `e.E1`.) So, I removed the test (and added an `Assert` as a sanity check.)